### ISSUE TITLE
VS 2019 when building a large project, even after a clean, projects fail either due to 'access denied' errors

### ISSUE THREAD

**Issue by JayCroghan** [dotnet/roslyn#45246]
**Version Used**: 16.5.4

**Steps to Reproduce**:

1. Clean build
2. Kill all MSBuild.exe tasks or it happens for even more files.
3. Build the solution

**Expected Behavior**: Solution builds.

**Actual Behavior**: At least one but usually multiple project fails to build with errors of 

"The file is locked by "System (4), MSBuild.exe (18228)""

"Access to the path X is denied"

"The process cannot access the file X because it is in use by another process"

It mostly happens for additional files like config files and content files for web apps but it happens for a variety of files including the binary targets such as dlls and exes.

The only way to get it working is try cleaning and rebuilding multiple times until they all succeed.

![errors](https://user-images.githubusercontent.com/1171148/84853840-ac42e380-b092-11ea-9ac5-6713307726f6.png)

**Comment by JayCroghan** [comment]
Sometimes after deleting the target files from the output directory subsequent build succeeds, sometimes it doesn't.

**Comment by JayCroghan** [comment]
Updated to 16.6.2 problem still persists.

**Comment by JayCroghan** [comment]
Using command line MSBuild also has the same problem.

**Comment by CyrusNajmabadi** [comment]
> Using command line MSBuild also has the same problem.

Perhaps some sort of antivirus or other issue on your box?

**Comment by JayCroghan** [comment]
I'll take a look at checking what process is locking it, from above it lists MSBuild in some instances.

I've request MSBuild to be added to the ignore list, should that be sufficient? devenv is already excluded.

**Comment by JayCroghan** [comment]
Once the build has finished handle is displaying no files open in my source or target foldes.

**Comment by JayCroghan** [comment]
Finally found a dll throwing an error with open handles when I try and build from the command line...

System & Microsoft.VisualStudio.Web.Host.exe

This was after a fresh reboot, I tried to build in visual studio and got some errors about access denied to log4net.config - then tried the command line.
![errors](https://user-images.githubusercontent.com/1171148/84878487-7c103a80-b0bc-11ea-8a28-10bb33e63fac.png)

**Comment by JayCroghan** [comment]
Disabling the virus scanner had no effect.

**Comment by CyrusNajmabadi** [comment]
Yeah, unfortunately that's not a Roslyn process at all. Maybe asp.net?  Try filling with them and see if they've ever seen this or know who owns this process.

**Comment by JayCroghan** [comment]
How do I fill with them? I’ll look more tomorrow, I’m sure some of the files cannot be used by this process. Some of the files that are getting the error are additional files set to copy with the build with winforms apps that have no relation whatsoever to the asp.net project so that doesn’t make any sense the sole asp.net project in the bunch of 455 projects is locking those unrelated files. I wonder how can I find out what’s locking them at the point in time of the error.

An example which is 100% you, if I create a new winforms app and run it, when I run a second time it says it’s in use and i need to wait a minute or two before running to avoid the error.

**Comment by CyrusNajmabadi** [comment]
> if I create a new winforms app and run it, when I run a second time it says it’s in use

We would have to know what exe is locking the binaries.  Perhaps you can use a tool like Process Hacker and see if it can log this information in some way?

> An example which is 100% you, 

I don't really see why WinForms would be anymore us than the web.host.exe case.  :-/

**Comment by CyrusNajmabadi** [comment]
Closing out.  This is not a roslyn issue.

**Comment by rikinpatel13** [comment]
@JayCroghan 
I am also face same issue, any solution?
